single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/styles/iidm/HighlightLineStateStyleProvider.java

```diff
@@ -46,38 +44,79 @@ private Optional<String> getHighlightLineStateStyle(Graph graph, Edge edge) {
+        } else if (n.getFeeder() instanceof FeederWithSides) {
+            FeederWithSides feederWs = (FeederWithSides) n.getFeeder();
+            Map<NodeSide, Boolean> connectionStatus = connectionStatus(n);
+            NodeSide side = null;
+            NodeSide otherSide = null;
+            if (feederWs.getFeederType() == FeederType.BRANCH || feederWs.getFeederType() == FeederType.TWO_WINDINGS_TRANSFORMER_LEG) {
+                side = feederWs.getSide();
                 otherSide = getOtherSide(side);
+                if (ComponentTypeName.LINE.equals(n.getComponentType())) {
+                    side = Boolean.TRUE.equals(connectionStatus.get(side)) ? side : otherSide;
+                    otherSide = getOtherSide(side);
+                }
+            } else if (feederWs.getFeederType() == FeederType.THREE_WINDINGS_TRANSFORMER_LEG) {
+                String idVl = graph.getVoltageLevelInfos(n).getId();
+                ThreeWindingsTransformer transformer = network.getThreeWindingsTransformer(n.getEquipmentId());
+                if (transformer != null) {
+                    side = getTransformerSide(idVl, transformer);
+                }
+                otherSide = feederWs.getSide();
             }
-        } else if (feederWs.getFeederType() == FeederType.THREE_WINDINGS_TRANSFORMER_LEG) {
-            String idVl = graph.getVoltageLevelInfos(n).getId();
-            ThreeWindingsTransformer transformer = network.getThreeWindingsTransformer(n.getEquipmentId());
-            if (transformer != null) {
-                side = getTransformerSide(idVl, transformer);
-            }
-            otherSide = feederWs.getSide();
+            return getFeederStateStyle(side, otherSide, connectionStatus);
+        } else {
+            return Optional.empty();
         }
+    }
 
-        return getFeederStateStyle(side, otherSide, connectionStatus);
+    protected Optional<String> getHighlightFeederStateStyleForInternal2WT(Node n1, Node n2) {
+        Internal2WTNode n2WT = (Internal2WTNode) (n1 instanceof Internal2WTNode ? n1 : n2);
+        if (!(n1 instanceof ConnectivityNode) && !(n2 instanceof ConnectivityNode)) {
+            return Optional.empty();
+        }
+        ConnectivityNode connectivityNode = (ConnectivityNode) (n1 instanceof ConnectivityNode ? n1 : n2);
+        String connectivityNodeId = connectivityNode.getId();
+        int connectivityNodeIdLength = connectivityNodeId.length();
+        String busIdFromConnectivityNode = connectivityNodeId.substring(connectivityNodeIdLength - 11, connectivityNodeIdLength - 8);
+        TwoWindingsTransformer internal2WT = (TwoWindingsTransformer) network.getIdentifiable(n2WT.getEquipmentId());
+        Terminal terminal1 = internal2WT.getTerminal1();
+        Terminal terminal2 = internal2WT.getTerminal2();
+        if (terminal1.getBusBreakerView().getConnectableBus().getId().equals(busIdFromConnectivityNode)) {
```

Comment:
I'm not sure it would work also in node/breaker or at least it would recompute buses for nothing